client/components/site-title/index.jsx

```diff
@@ -54,9 +70,9 @@ class SiteTitleControl extends React.Component {
 						disabled={ disabled }
 						id="blogname"
 						onChange={ this.onChangeSiteTitle }
-						required
 						value={ blogname }
 					/>
+					{ isTitleRequired ? this.renderValidation( blogname ) : null }
```

Comment:
I generally try to avoid `this.render...()` methods -- I find them sometimes distracting since they tempt me to scroll around, losing focus more easily.

I think we might be able to simply inline `<FormInputValidation isError={ isBlognameRequired && ! blogname } text={ translate( 'Required field.' ) />` here -- IMO that also reads self-explanatory enough, what do you think?

Reply:
(Might need to still wrap with `{ isTitleRequired && ... }`)